Compiler suggests #[derive(Clone, Eq + Clone)]

Let's say I want my struct to be comparable with anything that can be converted into the struct itself, but the actual comparison requires `Clone`:

```rust
use core::cmp::PartialEq;

#[derive(Clone, Eq)]
pub struct Struct<T>(T);

impl<T, U> PartialEq<U> for Struct<T>
where
    U: Into<Struct<T>> + Clone
{
    fn eq(&self, other: &U) -> bool {
        todo!()
    }
}

```

([Playground](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=e33b9eff546315e8260792228c183684))

Errors:

```
   Compiling playground v0.0.1 (/playground)
error[E0277]: the trait bound `T: Clone` is not satisfied
   --> src/lib.rs:3:17
    |
3   | #[derive(Clone, Eq)]
    |                 ^^ the trait `Clone` is not implemented for `T`
    |
    = note: required because of the requirements on the impl of `Clone` for `Struct<T>`
    = note: required because of the requirements on the impl of `PartialEq` for `Struct<T>`
    = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider further restricting this bound
    |
3   | #[derive(Clone, Eq + Clone)]
    |                    ^^^^^^^

error: aborting due to previous error

For more information about this error, try `rustc --explain E0277`.
error: could not compile `playground`

To learn more, run the command again with --verbose.

```

--- a/src/test/ui/suggestions/derive-clone-for-eq.fixed
+++ b/src/test/ui/suggestions/derive-clone-for-eq.fixed
@@ -1,0 +1,18 @@
+// run-rustfix
+// https://github.com/rust-lang/rust/issues/79076
+
+use std::cmp::PartialEq;
+
+#[derive(Clone, Eq)] //~ ERROR [E0277]
+pub struct Struct<T: std::clone::Clone>(T);
+
+impl<T: Clone, U> PartialEq<U> for Struct<T>
+where
+    U: Into<Struct<T>> + Clone
+{
+    fn eq(&self, _other: &U) -> bool {
+        todo!()
+    }
+}
+
+fn main() {}

--- a/src/test/ui/suggestions/derive-clone-for-eq.rs
+++ b/src/test/ui/suggestions/derive-clone-for-eq.rs
@@ -1,0 +1,18 @@
+// run-rustfix
+// https://github.com/rust-lang/rust/issues/79076
+
+use std::cmp::PartialEq;
+
+#[derive(Clone, Eq)] //~ ERROR [E0277]
+pub struct Struct<T>(T);
+
+impl<T: Clone, U> PartialEq<U> for Struct<T>
+where
+    U: Into<Struct<T>> + Clone
+{
+    fn eq(&self, _other: &U) -> bool {
+        todo!()
+    }
+}
+
+fn main() {}

--- a/src/test/ui/suggestions/derive-clone-for-eq.stderr
+++ b/src/test/ui/suggestions/derive-clone-for-eq.stderr
@@ -1,0 +1,25 @@
+error[E0277]: the trait bound `T: Clone` is not satisfied
+  --> $DIR/derive-clone-for-eq.rs:6:17
+   |
+LL | #[derive(Clone, Eq)]
+   |                 ^^ the trait `Clone` is not implemented for `T`
+   |
+note: required because of the requirements on the impl of `PartialEq` for `Struct<T>`
+  --> $DIR/derive-clone-for-eq.rs:9:19
+   |
+LL | impl<T: Clone, U> PartialEq<U> for Struct<T>
+   |                   ^^^^^^^^^^^^     ^^^^^^^^^
+note: required by a bound in `Eq`
+  --> $SRC_DIR/core/src/cmp.rs:LL:COL
+   |
+LL | pub trait Eq: PartialEq<Self> {
+   |               ^^^^^^^^^^^^^^^ required by this bound in `Eq`
+   = note: this error originates in the derive macro `Eq` (in Nightly builds, run with -Z macro-backtrace for more info)
+help: consider restricting type parameter `T`
+   |
+LL | pub struct Struct<T: std::clone::Clone>(T);
+   |                    +++++++++++++++++++
+
+error: aborting due to previous error
+
+For more information about this error, try `rustc --explain E0277`.

